runtime/README.md

```diff
@@ -692,107 +791,628 @@ Command Code: `0x434B_4558` ("CKEX")
+
+*Table: `CM_AES_GCM_DECRYPT` output arguments*
+| **Name**       | **Type** | **Description**                           |
+| -------------- | -------- | ----------------------------------------- |
+| chksum         | u32      |                                           |
+| fips_status    | u32      | FIPS approved or an error                 |
+| context size   | u32      | Can be 0 if this is the final message     |
+| context        | u8[...]  |                                           |
+| plaintext size | u32      | Can be 0 if this is not the final message |
+| plaintext      | u8[...]  |                                           |
+
+The encrypted and authenticated context's internal structure will be:
+
+*Table: `CM_AES_GCM_DECRYPT` internal context*
+| **Name**        | **Type** | **Description** |
+| --------------- | -------- | --------------- |
+| tag size        | u32      |                 |
+| last length     | u32      |                 |
+| last iv         | u8[16]   |                 |
+| last GHASH      | u8[16]   |                 |
+| last ciphertext | u8[16]   |                 |
+
+
+### CM_ECDH_GENERATE
+
+This computes the first half of an Elliptic Curve Diffie-Hellman exchange to compute an ephemeral shared key pair with another party.
+
+Currently only supports the NIST P-384 curve.
+
+The returned context must be passed to the `CM_ECDH_FINISH` command. The context contains the (encrypted) secret coefficient.
+
+The returned exchange data format is the concatenation of the x- and y-coordinates of the public point.
+
+Command Code: `0x434D_4547` ("CMEG")
+
+*Table: `CM_ECDH_GENERATE` input arguments*
+| **Name**    | **Type** | **Description**      |
+| ----------- | -------- | -------------------- |
+| chksum      | u32      |                      |
+| curve/flags | u32      | Must be 0. Reserved. |
+
+*Table: `CM_ECDH_GENERATE` output arguments*
+| **Name**           | **Type** | **Description**                       |
+| ------------------ | -------- | ------------------------------------- |
+| chksum             | u32      |                                       |
+| fips_status        | u32      | FIPS approved or an error             |
+| context size       | u32      | size of context                       |
+| context            | u8[...]  | Used as the input to `CM_ECDH_FINISH` |
+| exchange data      | u8[96]   | i.e., the public point                |
+
+*Table: `CM_ECDH_GENERATE` / `CM_ECDH_FINISH` internal context*
+| **Name**           | **Type** | **Description** |
+| ------------------ | -------- | --------------- |
+| Secret coefficient | u8[48]   |                 |
+
+### CM_ECDH_FINISH
+
+This computes the second half of an Elliptic Curve Diffie-Hellman exchange.
+
+Currently only supports the NIST P-384 curve.
+
+The context must be passed from the `CM_ECDH_GENERATE` command.
+
+The incoming exchange data MUST be the concatenation of the x- and y- coordinates of the other side's public point.
```

Comment:
How are the coordinates encoded? Big-endian?

Reply:
Yes, big-endian.